### PR TITLE
[JENKINS-68205] Strange line breaks on Firefox 91.8.0esr when Build History uses wbr within display: inline-flex (regression in 2.321)

### PR DESCRIPTION
**Pull request by Langer0416**
See [JENKINS-68205](https://issues.jenkins.io/browse/JENKINS-68205).

before: 

before

after:

after

The table won't  be affected if your job don't have the name like this, e.g:

`XXX » xxxxxxxxxxxx-xxxxxxxx-xxxx » xxxxxxx1111FXXXXXXX-1234_xxxxxxxxxxxxXXXxxxxXXX » xxxxxxxxxxxx-xxxxxxxx-xxxx » xxxxxxx1111FXXXXXXX-1234_xxxxxxxxxxxxXXXxxxx`

### Proposed changelog entries

* Job's name won't seperate randomly in the table on Firefox.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

### CODE REVIEW

**Comment by janfaracik** [comment]
Haven't tested, but might be possible to just wrap the text inside of the link in a span to fix it.

**Comment by Langer0416** [comment]
@janfaracik , thanks for your reply and approach. I have fixed it as your approach, please have the  look at it.

### Before:
Before

### After:
After

**Comment by janfaracik** [comment]
Looks good to me :)

**Review by janfaracik — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]

**Comment by basil** [comment]
I cleaned up the indentation in commit d4158c18c6. @Langer0416 Could you please retest and confirm JENKINS-68205 is fixed in the latest revision? After that, I think this PR will be ready for merge. Thanks!

**Comment by Langer0416** [comment]
> I cleaned up the indentation in commit [d4158c1](https://github.com/jenkinsci/jenkins/commit/d4158c18c6c02bbee760a4b0728b58ad556c1783). @Langer0416 Could you please retest and confirm JENKINS-68205 is fixed in the latest revision? After that, I think this PR will be ready for merge. Thanks!

Hi, @basil. I have tested it in the latsest version (including "indentation") and it still works well on this PR.

**Comment by KalleOlaviNiemitalo** [comment]
Adding `span`  fixes the line breaks within the job link, but how about the build number in the build history table…

Original:

```HTML
<a class="jenkins-table__link model-link inside" href="/jenkins/job/XXXX/job/Xxxxx-Xxxxxx/job/XX-1111/">Xx Xxxx » Xxxxx<wbr>-Xxxxxx » XX<wbr>-1111</a>
```

Firefox 91.9.0esr, hovering over the job link:

Original strangely broken job link in the build history table

Adding `span` within the `a` makes the job link look nicer, but the table still wastes some vertical space by having a line break between the job link and the build number:

Sane job link after adding span in the build history table

If, instead of adding `span`, I remove the CSS rule

```CSS
.jenkins-table__link, .jenkins-table .sortheader {
    display: inline-flex;
}
```

then there is no longer a line break between the job link and the build number, saving vertical space:

Job link and build number at the same line

The shaded backgrounds overlap in all three cases. In the last case though, the shaded background of the job link is not large enough for the entire link. I guess adding `span` is the best fix, then.

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Comment by basil** [comment]
Since this changes the markup, I fear there may be a risk to Acceptance Test Harness (ATH). Let's try and confirm that this PR doesn't cause any regressions there prior to merging.

File: core/src/main/resources/lib/hudson/buildListTable.jelly
Comment (Langer0416) [inline core/src/main/resources/lib/hudson/buildListTable.jelly:48]: This part improves the break lines in the "Build history page".

### Before
Before

### After
After

File: core/src/main/resources/lib/hudson/buildListTable.jelly
Comment (Langer0416) [inline core/src/main/resources/lib/hudson/buildListTable.jelly:49]: This part improves break lines in the "Build History page".
### Before
Before
### After
After

**Comment by basil** [comment]
[PR build 21](https://ci.jenkins.io/job/Core/job/jenkins/job/PR-6542/21/) produced [incremental 2.349-rc32401.9d9fa_a_9d7c40](https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/main/cli/2.349-rc32401.9d9fa_a_9d7c40/).

@NotMyFault was kind enough to submit https://github.com/jenkinsci/acceptance-test-harness/pull/773 to test this incremental build in the Acceptance Test Harness (ATH). [PR build 3](https://ci.jenkins.io/job/Core/job/acceptance-test-harness/job/PR-773/3/) showed failures in `plugins.JiraPluginTest.jira_ticket_gets_updated_with_a_build_link`, `plugins.JobDslPluginTest.should_approve_administrator_script_automatically`, `plugins.JobDslPluginTest.should_disallow_importing_groovy_classes_when_script_security_enabled`, and `plugins.OwnershipPluginTest.explicitly_set_ownership`. Those same test failures are also present in [trunk build 452](https://ci.jenkins.io/job/Core/job/acceptance-test-harness/job/master/452/), so we can conclude that this PR introduces no new test failures. :thumbsup:

@Langer0416 To be safe, can you please test the same incremental build in `jenkinsci/bom`? To do so, file a draft PR updating the core version to the incremental build referenced above in `jenkinsci/bom/sample-plugin/pom.xml` as in e.g. https://github.com/jenkinsci/bom/pull/1113. Once we are confident that this PR does not introduce any test failures in `jenkinsci/bom`, I think it will be ready for merge.

**Comment by NotMyFault** [comment]
> @Langer0416 To be safe, can you please test the same incremental build in `jenkinsci/bom`? To do so, file a draft PR updating the core version to the incremental build referenced above in `jenkinsci/bom/sample-plugin/pom.xml` as in e.g. [jenkinsci/bom#1113](https://github.com/jenkinsci/bom/pull/1113). Once we are confident that this PR does not introduce any test failures in `jenkinsci/bom`, I think it will be ready for merge.

Bom is green too.

**Comment by basil** [comment]
Thank you Alex, it is inspiring to see members of our community supporting each other in pursuit of a common goal.

**Comment by Langer0416** [comment]
> > @Langer0416 To be safe, can you please test the same incremental build in `jenkinsci/bom`? To do so, file a draft PR updating the core version to the incremental build referenced above in `jenkinsci/bom/sample-plugin/pom.xml` as in e.g. [jenkinsci/bom#1113](https://github.com/jenkinsci/bom/pull/1113). Once we are confident that this PR does not introduce any test failures in `jenkinsci/bom`, I think it will be ready for merge.
> 
> Bom is green too.

@NotMyFault, thank you very much for your assistances.

**Pull request merged**
